File: backend/app/types/gl_account_rules.py
from pydantic import BaseModel
from typing import Optional


class CreateAccountRuleRequest(BaseModel):
	dimension_id: int
	is_required: bool = False
	parent_value_id: Optional[int] = None
	allowed_value_ids: list[int] = []


class UpdateAccountRuleRequest(BaseModel):
	is_required: bool


class SetRuleValuesRequest(BaseModel):
	value_ids: list[int]

File: frontend/app/lib/types/gl_account_rules.ts
export interface AccountRule {
	id: number;
	dimension_id: number;
	is_required: boolean;
	parent_value_id: number | null;
	allowed_value_ids: number[];
}

export interface AccountRulesData {
	account_id: number;
	rules: AccountRule[];
}

export interface CreateAccountRuleRequest {
	dimension_id: number;
	is_required: boolean;
	parent_value_id: number | null;
	allowed_value_ids: number[];
}

export interface UpdateAccountRuleRequest {
	is_required: boolean;
}

export interface SetRuleValuesRequest {
	value_ids: number[];
}
